satellite/satellitedb/orders.go

```diff
@@ -125,20 +129,44 @@ func (db *ordersDB) UpdateBucketBandwidthInline(ctx context.Context, bucketID []
 }
 
 // UpdateStoragenodeBandwidthAllocation updates 'allocated' bandwidth for given storage node
-func (db *ordersDB) UpdateStoragenodeBandwidthAllocation(ctx context.Context, storageNode storj.NodeID, action pb.PieceAction, amount int64, intervalStart time.Time) (err error) {
+func (db *ordersDB) UpdateStoragenodeBandwidthAllocation(ctx context.Context, storageNodes []storj.NodeID, action pb.PieceAction, amount int64, intervalStart time.Time) (err error) {
 	defer mon.Task()(&ctx)(&err)
-	statement := db.db.Rebind(
-		`INSERT INTO storagenode_bandwidth_rollups (storagenode_id, interval_start, interval_seconds, action, allocated, settled)
-		VALUES (?, ?, ?, ?, ?, ?)
-		ON CONFLICT(storagenode_id, interval_start, action)
-		DO UPDATE SET allocated = storagenode_bandwidth_rollups.allocated + ?`,
-	)
-	_, err = db.db.ExecContext(ctx, statement,
-		storageNode.Bytes(), intervalStart, defaultIntervalSeconds, action, uint64(amount), 0, uint64(amount),
-	)
-	if err != nil {
-		return err
+
+	switch t := db.db.Driver().(type) {
+	case *sqlite3.SQLiteDriver:
+		statement := db.db.Rebind(
+			`INSERT INTO storagenode_bandwidth_rollups (storagenode_id, interval_start, interval_seconds, action, allocated, settled)
+			VALUES (?, ?, ?, ?, ?, ?)
```

Comment:
If you want, we could add a row of `, (?, ?, ?, ?, ?, ?)` here for each storageNode, and perform all the inserts at the same time in the same statement. Sqlite supports that ok.

like

    INSERT INTO storagenode_bandwidth_rollups (storagenode_id, interval_start, interval_seconds, action, allocated, settled)
    VALUES (?, ?, ?, ?, ?, ?)
    , (?, ?, ?, ?, ?, ?)
    , (?, ?, ?, ?, ?, ?)
    , (?, ?, ?, ?, ?, ?)
    ON CONFLICT (storagenode_id, interval_start, action)
    DO UPDATE SET allocated = storagenode_bandwidth_rollups.allocated + excluded.allocated

With sqlite being an in-memory db where round-trips aren't very costly, this might not have any meaningful effect on performance (and we may not particularly care about performance here anyway), but at least it would be more obvious that the inserts are meant to be done all-at-once, in the same transaction?

Reply:
They aren't required to be done in the same transaction and the values change would complicate the code here quite a bit. So, I would prefer keep the code simpler for sqlite.

Reply:
Sure, it's your call.

Just for consideration though, the more the semantic details diverge between the sqlite side (which powers most of our tests) and the postgresql side (mostly used in production), the effectiveness of our tests suffers. I don't see how it could matter here, so I won't push very hard, but "we do these few hundred things in a single transaction in production but many separate transactions in test" is pretty scary to me.

Reply:
Most of the tests do run against sqlite and postgres both in Jenkins. i.e. `testplanet.Run` and `satellitedbtest.Run` iterate over database configurations.

There's only a difference, if you haven't setup testing postgres locally.